### ISSUE TITLE
NAs in a blocking variable

### ISSUE THREAD

**Issue by benthestatistician** [markmfredrickson/optmatch#206]
If a stratifying variable provided to `exactMatch()` or `strata()` contains an NA, I'd expect the case with an NA simply to go unmatched.  However, in released version 0.9-13 the effect of such an NA seems to be to erase traces of that case at the distance-making stage, causing problems.  In my script:

```r
with(counties, table(mass_cnty, is.na(ps2_strat2)))         

## mass_cnty FALSE TRUE
##     FALSE   917 2210
##     TRUE     14    0

dim(match_on(mass_cnty ~ ppty0 + log10(a20_64_cnt), data=counties))
##[1]   14 3127

dim(exactMatch(mass_cnty ~ ps2_strat2, data=counties) )
## [1]  14 917

dim(match_on(mass_cnty ~ ppty0 + log10(a20_64_cnt) + strata(ps2_strat2), data=counties))
## Error in makedist(z, data, compute_mahalanobis, within) : 
##  Row and column names of within must match those of the data.
```
Interestingly, at the same time
```r
dim(match_on(mass_cnty ~ ps2_strat2, data=counties))
## [1]   14 3127
```

I think the remedy is to persuade `exactMatch()` and friends to return a BISM of the same dimension as `match_on()`'s output.

**Comment by josherrickson** [comment]
Just to note, `survival::strata` contains the `na.group` option which, if `TRUE`, treats `NA` as another strata.

**Comment by benthestatistician** [comment]
1.  I'm wary of making NAs into a stratum, as `survival::strat(<...>, na.group=TRUE)` would do; that's the old behavior that led to #203, as well as to similar bugs in the distant past. 
2. When we get to fixing this -- 

- [ ] have the NEWS entry write up our fix for #189 as well 
 
-- it being similar to this  issue from the user perspective.

**Comment by benthestatistician** [comment]
Commit 9928647, on branch issue206-NAs-in-blocking, fixes this problem while introduces a few others.  They seem fixable, but I'm leaving that for another day.
